File: js/src/test/base/test.groupBy.js
// AUTO_TRANSPILE_ENABLED
import ccxt from '../../../ccxt.js';
import testSharedMethods from '../Exchange/base/test.sharedMethods.js';
function testGroupBy() {
    const exchange = new ccxt.Exchange({
        'id': 'sampleexchange',
    });
    const sampleArray = [
        { 'foo': 'a' },
        { 'foo': 'b' },
        { 'foo': 'c' },
        { 'foo': 'b' },
        { 'foo': 'c' },
        { 'foo': 'c' },
    ];
    const currentValue = exchange.groupBy(sampleArray, 'foo');
    const storedValue = {
        'a': [{ 'foo': 'a' }],
        'b': [{ 'foo': 'b' }, { 'foo': 'b' }],
        'c': [{ 'foo': 'c' }, { 'foo': 'c' }, { 'foo': 'c' }],
    };
    testSharedMethods.assertDeepEqual(exchange, undefined, 'testGroupBy', currentValue, storedValue);
}
export default testGroupBy;

File: julia/Ccxt/test/base/test.groupBy.jl
using Test
using Ccxt
function testGroupBy()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    sampleArray = [Dict{Symbol, Any}(
        Symbol("foo") => "a"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "b"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "c"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "b"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "c"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "c"
    )];
    currentValue = groupBy(exchange, sampleArray, "foo");
    storedValue = Dict{Symbol, Any}(
        Symbol("a") => [Dict{Symbol, Any}(
        Symbol("foo") => "a"
    )],
        Symbol("b") => [Dict{Symbol, Any}(
        Symbol("foo") => "b"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "b"
    )],
        Symbol("c") => [Dict{Symbol, Any}(
        Symbol("foo") => "c"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "c"
    ), Dict{Symbol, Any}(
        Symbol("foo") => "c"
    )]
    );
    assertDeepEqual(testSharedMethods, exchange, nothing, "testGroupBy", currentValue, storedValue);
end
